core/services/download_service.py

```
@@ -0,0 +1,86 @@
+"""
+Download Service Implementation
+
+Manages download operations and progress tracking.
+"""
+
+import logging
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Callable
+from datetime import datetime
```

Ruff (0.11.9)

`datetime.datetime` imported but unused

Remove unused import: `datetime.datetime`

(F401)